core/charm/repository/charmhub.go

```diff
@@ -44,7 +44,58 @@ func NewCharmHubRepository(logger Logger, chClient CharmHubClient) *CharmHubRepo
 // ResolveWithPreferredChannel defines a way using the given charm URL and
 // charm origin (platform and channel) to locate a matching charm against the
 // Charmhub API.
-//
+func (c *CharmHubRepository) ResolveWithPreferredChannel(charmURL *charm.URL, argOrigin corecharm.Origin) (*charm.URL, corecharm.Origin, []string, error) {
```

Comment:
Why do we still have this `ResolveWithPrefferedChannel`, isn't it the same with `ResolveForDeploy` (if it returned the `resolvedOrigin` too) ?

Reply:
ResolveWithPreferredChannel is needed for use by bundles and older juju 3.x clients. The main difference between the two is the data returned. ResolveForDeploy has no need for supported series but does need the essential metadata from the charmhub calls made to resolve the charm.

The resolvedOrigin is contained in ResolveForDeploy response, it's down in the EssentialMetadata